urllib3/util/retry.py

```diff
@@ -273,10 +273,18 @@ def _is_read_error(self, err):
         """
         return isinstance(err, (ReadTimeoutError, ProtocolError))
 
+    def _is_method_retryable(self, method):
+        """ Is this method retryable? (Based on method whitelist)
+        """
+        if self.method_whitelist and method.upper() not in self.method_whitelist:
+            return False
+
+        return True
+
     def is_retry(self, method, status_code, has_retry_after=False):
         """ Is this method/status code retryable? (Based on method/codes whitelists)
```

Comment:
@Lukasa True, how about for `_is_method_retryable`: """ Checks if a given HTTP method should be retried upon, depending if it is included on the method whitelist."""

And for `is_retry`: """ Is this method/status code retryable? (Based on whitelists)"""

I could also change the one for `is_retry` to include the fact that it also checks variables such as `total`, `respect_retry_after_header`, and `has_retry_after`.